src/checkqueue.h

```diff
@@ -90,13 +92,13 @@ class CCheckQueue
                         nTotal--;
                         bool fRet = fAllOk;
                         // reset the status for new work later
-                        if (fMaster)
-                            fAllOk = true;
+                        fAllOk = true;
                         // return the current status
                         return fRet;
                     }
                     nIdle++;
-                    cond.wait(lock); // wait
+                    cond.wait(lock, [&] { return interrupted || !queue.empty() || fMaster && nTodo == 0; }); // wait
```

Comment:
I think you could assert(! (fMaster && nTodo == 0)) here.

If fMaster were *ever* true here, we would deadlock.

Hence, there is no point to check that...